internal/store/selector/label.go

```
@@ -0,0 +1,309 @@
+package selector
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/flightctl/flightctl/internal/flterrors"
+	"github.com/flightctl/flightctl/pkg/k8s/selector"
+	"github.com/flightctl/flightctl/pkg/k8s/selector/labels"
+	"github.com/flightctl/flightctl/pkg/k8s/selector/selection"
+	"github.com/flightctl/flightctl/pkg/queryparser"
+	"github.com/flightctl/flightctl/pkg/queryparser/sqljsonb"
+)
+
+type LabelSelector struct {
+	field    *SelectorField
+	parser   queryparser.Parser
+	selector selector.Selector
+}
+
+// NewLabelSelectorFromMapOrDie creates a LabelSelector from a map of labels
+// with an optional invert flag. It panics if the creation fails.
+//
+// Parameters:
+//
+//	labels - A map where keys are label names and values are label values.
+//	invert - (Optional) If true, inverts the operator to "!=" instead of "=".
+//
+// Example:
+//
+//	labels := map[string]string{"env": "prod", "tier": "backend"}
+//	selector := NewLabelSelectorFromMapOrDie(labels)
+//	// selector represents: "env=prod,tier=backend"
+func NewLabelSelectorFromMapOrDie(labels map[string]string, invert ...bool) *LabelSelector {
```

1. Why panic instead of returning an error?
2. I get that you want "invert" to be optional, but it's not clear because it looks I can/should pass multiple values.  I don't feel too strongly about it, and I saw that it's documented above as optional, but I found it confusing.

---

1. ParseSelectorOrDie is a practice used by the Kubernetes package. It simplifies specific cases and is primarily utilized in tests or scenarios where the query is created by the backend, ensuring no potential mistakes from end-user input. For safer handling of errors, we also provide the NewAnnotationSelectorFromMap function, which processes errors accordingly.

2. I'll fix the 'invert'